ibm/resource_ibm_pi_instance.go

```diff
@@ -268,41 +211,55 @@ func resourceIBMPIInstance() *schema.Resource {
 				Description: "PI instance image id",
 			},
 			helpers.PIInstanceProcessors: {
-				Type:        schema.TypeFloat,
-				Required:    true,
-				Description: "Processors count",
+				Type:          schema.TypeFloat,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{PISAPInstanceProfileID},
+				Description:   "Processors count",
 			},
 			helpers.PIInstanceName: {
 				Type:        schema.TypeString,
 				Required:    true,
 				Description: "PI Instance name",
 			},
 			helpers.PIInstanceProcType: {
-				Type:         schema.TypeString,
-				Required:     true,
-				ValidateFunc: validateAllowedStringValue([]string{"dedicated", "shared", "capped"}),
-				Description:  "Instance processor type",
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ValidateFunc:  validateAllowedStringValue([]string{"dedicated", "shared", "capped"}),
+				ConflictsWith: []string{PISAPInstanceProfileID},
+				Description:   "Instance processor type",
 			},
 			helpers.PIInstanceSSHKeyName: {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
 				Description: "SSH key name",
 			},
 			helpers.PIInstanceMemory: {
-				Type:        schema.TypeFloat,
-				Required:    true,
-				Description: "Memory size",
+				Type:          schema.TypeFloat,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{PISAPInstanceProfileID},
+				Description:   "Memory size",
+			},
+			PISAPInstanceProfileID: {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{helpers.PIInstanceProcessors, helpers.PIInstanceMemory, helpers.PIInstanceProcType, helpers.PIInstanceSystemType},
+				Description:   "SAP Profile ID for the amount of cores and memory",
 			},
 			helpers.PIInstanceSystemType: {
-				Type:         schema.TypeString,
-				Required:     true,
-				ValidateFunc: validateAllowedStringValue([]string{"s922", "e880", "e980"}),
-				Description:  "PI Instance system type",
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ValidateFunc:  validateAllowedStringValue([]string{"s922", "e880", "e980"}),
+				ConflictsWith: []string{PISAPInstanceProfileID},
+				Description:   "PI Instance system type",
 			},
 			helpers.PIInstanceReplicants: {
-				Type:        schema.TypeFloat,
+				Type:        schema.TypeInt,
 				Optional:    true,
-				Default:     1.0,
```

Comment:
Changing default value might effect existing users config..

Reply:
Yes but we need it as int for SAP also it makes sense the replication factor cannot be a float value. Logically don't think it will make any difference for existing users.